### ISSUE TITLE
GS63VR Wave mode doesn't work

### ISSUE THREAD

**Issue by AbsyntheSyne** [Gibtnix/MSIKLM#17]
Wave mode doesn't work. It used to work previously, I don't remember what commit. I type in `sudo msiklm wave` and it does nothing.

**Comment by AbsyntheSyne** [comment]
Yes I do. It refuses to work for me.

**Comment by Gibtnix** [comment]
Strange... could you please try all of the following commands if any works (there are different ways how the program works, so maybe please try all three)
- sudo msiklm green wave
- sudo msiklm green,blue,red high wave
- sudo msiklm green high wave
If none of these work, I'll prepare a special version for you to try to locate the problem.

**Comment by AbsyntheSyne** [comment]
* no

* yes

* yes

**Comment by Gibtnix** [comment]
First it seems that explicitly supplying the brightness 'high' is important. Internally the default value is 'rgb' such that the rgb-color setting is used (cf. set_color() in msiklm.c), you can switch this to predefined colors by using low, medium or high instead. So the latter two commands I asked you to test are especially interesting as they use the explicit brightness (and thus, are switching to predefined mode) and it seems to work so far. So I would conclude that your keyboard requires to use the predefined color mode.

But I cannot explain why 'sudo msiklm red,green,blue high wave' does not work while 'sudo msiklm red,blue,green high wave' does - the commands are equivalent with respect to mode setting as long as you explicitly supply 'high' brightness. Color parsing also works, I verified this once more.

**Comment by AbsyntheSyne** [comment]
Yeah I don't know what's up, but it's really confusing. Any ideas why this is happening?
